finesse/gui/temp_control.py

```diff
@@ -41,138 +34,147 @@ def _create_controls(self) -> QGridLayout:
             QGridLayout: The layout containing the figure.
         """
         layout = QGridLayout()
+        self._btns = {"hot": QPushButton("Hot BB"), "cold": QPushButton("Cold BB")}
+        self._btns["hot"].clicked.connect(  # type: ignore
+            partial(self._btn_pressed, name="hot")
+        )
+        self._btns["cold"].clicked.connect(  # type: ignore
+            partial(self._btn_pressed, name="cold")
+        )
         self._create_figure()
 
-        layout.addWidget(self._canvas, 0, 0)
+        layout.addWidget(self._btns["hot"], 0, 0)
+        layout.addWidget(self._btns["cold"], 1, 0)
+        layout.addWidget(self._canvas, 0, 1, 2, 1)
 
         return layout
 
     def _create_figure(self) -> None:
-        """Creates the matplotlib figure to be contained within the panel.
-
-        Returns:
-            None
-        """
-        self._figure, self._ax1 = plt.subplots()
+        """Creates the matplotlib figure to be contained within the panel."""
+        self._figure, ax = plt.subplots()
+        self._ax = {"hot": ax}
         self._canvas = FigureCanvasQTAgg(self._figure)
 
-        t = [1040, 1050, 1060, 1070, 1080, 1090, 1100]
-        hot_bb_temp = [55, 57.5, 60, 62.5, 65, 67.5, 70]
-        cold_bb_temp = [1, 1.1, 1.15, 1.2, 1.3, 1.4, 1.5]
+        # Placeholder callback to test updating of figure
+        self._canvas.mpl_connect("button_press_event", self._update_figure)
+
+        self._figure_num_pts = 10
+        t = [None] * self._figure_num_pts
+        hot_bb_temp = [None] * self._figure_num_pts
+        cold_bb_temp = [None] * self._figure_num_pts
+
+        colours = plt.rcParams["axes.prop_cycle"]
+        hot_colour = colours.by_key()["color"][0]
+        cold_colour = colours.by_key()["color"][1]
 
-        self._bb_hot_line = self._ax1.plot(t, hot_bb_temp, color=[0, 1, 0])
-        self._ax1.set_xlabel("")
-        self._ax1.set_ylabel("HOT BB", color=[0, 1, 0])
-        self._ax1.set_xlim([1045, 1101.4])
-        self._ax1.set_ylim([20, 80])
+        self._ax["hot"].plot(t, hot_bb_temp, color=hot_colour)
```

Comment:
I'm not sure why you're plotting `hot_bb_temp` and `cold_bb_temp` when they've been set to all `None`s. Surely that means nothing will be plotted...? If so, you can just drop this call to `plot()` and not bother setting `hot_bb_temp` and `cold_bb_temp` above.

Reply:
Ah, now I see you're actually for these `None`s in `_update_figure` below. The way I'd do this is to just plot zero points then in `_update_figure` you can just check this with `if xdata`.
 
```suggestion
        self._ax["hot"].plot(t, (), color=hot_colour)
```

Reply:
This is just so that the plot always shows a fixed (maximum) number of points, which helps it update nicely. The `None`s get overwritten as more points are collected, rather than appending to a list. 

But yeah, I'm not sure whether it's better/preferable to have a fixed number of points or show a fixed time interval so this may change anyway.